### PR TITLE
Created classification homework

### PR DESCRIPTION
**Pull request by ThinamXx**

### CODE REVIEW

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+* `'households'`,
+* `'median_income'`,
+* `'median_house_value'`
+* `'ocean_proximity'`,
+
+Select only them and fill in the missing values with 0.
+
+### Question 1
+
+What is the most frequent observation (mode) for the column 'ocean_proximity'?
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: 'ocean_proximity' -> `ocean_proximity`

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+
+## Split the data
+* Create a new column `rooms_per_household` by dividing the column `total_rooms` by the column `households` from dataframe. 
+* Create a new column `bedrooms_per_room` by dividing the column `total_bedrooms` by the column `total_rooms` from dataframe. 
+* Create a new column `population_per_household` by dividing the column `population` by the column `households` from dataframe.
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: May be it's better to move these 3 lines to 'Features' section?

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+### Question 2
+
+* Create the [correlation matrix](https://www.google.com/search?q=correlation+matrix) for the numerical features of your train dataset.
+    - In a correlation matrix, you compute the correlation coefficient between every pair of features in the dataset.
+* What are the two features that have the biggest correlation in this dataset?
+
+### Make `median_house_value` binary
+* We need to turn the `median_house_value` variable from numeric into binary.
+* Let's create a variable `above_average` which is `1` if the median_house_value is above its mean value and 0 otherwise.
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: `median_house_value`, `0`

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+What is the most frequent observation (mode) for the column 'ocean_proximity'?
+
+
+## Split the data
+* Create a new column `rooms_per_household` by dividing the column `total_rooms` by the column `households` from dataframe. 
+* Create a new column `bedrooms_per_room` by dividing the column `total_bedrooms` by the column `total_rooms` from dataframe. 
+* Create a new column `population_per_household` by dividing the column `population` by the column `households` from dataframe. 
+* Split your data in train/val/test sets, with 60%/20%/20% distribution.
+* Use Scikit-Learn for that (the `train_test_split` function) and set the seed to 42.
+* Make sure that the target value ('median_house_value') is not in your dataframe.
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: `median_house_value`

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+* Calculate the mutual information score with the (binarized) price for the categorical variable that we have. Use the training set only.
+* What is the value of mutual information?
+* Round it to 2 decimal digits using `round(score, 2)`
+
+Options:
+- 0.163
+- 0.00001
+- 0.101
+- 0.15555
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: I don't know what the correct answer is here, but if we take only 2 decimal digits after the decimal point, it looks a bit confusing, as the first and last options are quite close. May be it's better to change one of them?

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+
+### Question 4
+
+* Now let's train a logistic regression
+* Remember that we have one categorical variables in the data. Include them using one-hot encoding.
```
Comment (SVizor42) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: ... categorical _variable_ in the data. Include _it_ using ...

**Comment by ThinamXx** [comment]
I have updated the bugs.

**Review by SVizor42 — APPROVED** [review]
I just added tiny changes. Now it looks good for me! Thanks, Thinam!

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+* Select only them and fill in the missing values with 0.
+* Create a new column `rooms_per_household` by dividing the column `total_rooms` by the column `households` from dataframe. 
+* Create a new column `bedrooms_per_room` by dividing the column `total_bedrooms` by the column `total_rooms` from dataframe. 
+* Create a new column `population_per_household` by dividing the column `population` by the column `households` from dataframe. 
+
+
+### Question 1
+
+What is the most frequent observation (mode) for the column `ocean_proximity`?
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md:38]: Can you please add options here? It'll be easier to evaluate the results later if we have options

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+
+### Split the data
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: Please add linebreak between headers and content

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+### Question 2
+
+* Create the [correlation matrix](https://www.google.com/search?q=correlation+matrix) for the numerical features of your train dataset.
+    - In a correlation matrix, you compute the correlation coefficient between every pair of features in the dataset.
+* What are the two features that have the biggest correlation in this dataset?
+
+
+### Make `median_house_value` binary
+* We need to turn the `median_house_value` variable from numeric into binary.
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md:68]: Extra linebreak here too

**Comment by alexeygrigorev** [comment]
Looks great! Thanks! Just a few comments and it's ready

I asked you already, but can you please rebase your fork to the current master? It's just a few git commands that you need to run, but it'll make pull requests a bit cleaner (not for this PR, but for future ones)

**Comment by ThinamXx** [comment]
Yes, I tried to rebase it to the current master.

**Comment by alexeygrigorev** [comment]
Just FYI - rebase and merge are different things. For your next PR, can you please [rebase your fork](https://gist.github.com/ravibhure/a7e0918ff4937c9ea1c456698dcd58aa)? (no need to do it for this one)

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+* Split your data in train/val/test sets, with 60%/20%/20% distribution.
+* Use Scikit-Learn for that (the `train_test_split` function) and set the seed to 42.
+* Make sure that the target value (`median_house_value`) is not in your dataframe.
+
+### Question 2
+
+* Create the [correlation matrix](https://www.google.com/search?q=correlation+matrix) for the numerical features of your train dataset.
+    - In a correlation matrix, you compute the correlation coefficient between every pair of features in the dataset.
+* What are the two features that have the biggest correlation in this dataset?
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md:57]: can we add options here too?

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+- 0.01
+- 0.1
+- 1
+- 10
+
+
+
+## Submit the results
+
+* Submit your results here: FORM LINK
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: It looks good, so please create a form. Thank you!

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+* You can submit your solution multiple times. In this case, only the last submission will be used 
+* If your answer doesn't match options exactly, select the closest one
+
+
+## Deadline
+
+The deadline for submitting is (will be updated soon), Timezone.
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md]: 26 September (Monday), 23:00 CEST

Comment (alexeygrigorev) [reply course-zoomcamp/cohorts/2022/03-classification/homework.md]: In general, it's usually one week

**Comment by ThinamXx** [comment]
I have updated the bugs raised

File: course-zoomcamp/cohorts/2022/03-classification/homework.md [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+- 0.01
+- 0.1
+- 1
+- 10
+
+
+
+## Submit the results
+
+* Submit your results here: https://forms.gle/MTVZ1wrHEiHYprQV7
```
Comment (alexeygrigorev) [inline course-zoomcamp/cohorts/2022/03-classification/homework.md:136]: Can you please use the form template? I see that some of the questions miss description (and they are present in the template)

**Comment by alexeygrigorev** [comment]
PR accepted - thanks! Just one last comment regarding the form - please use the template so the description of the fields is copied

**Comment by alexeygrigorev** [comment]
Since I squashed all the commits into one when merging - please also rebase your fork. Thanks!

**Pull request merged**